scripts/disk_check.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+What:
+    There have been cases, where disk turns RO due to kernel bug.
+    In RO state, system blocks new remote user login via TACACS.
+    This utility is to check & make transient recovery as needed.
+
+How:
+    check for RW permission. If RO, create writable overlay using tmpfs.
+
+    By default "/etc" & "/home" are checked and if in RO state, make them RW
+    using overlay on top of tmpfs.
+
+    Making /etc/ & /home as writable lets successful new remote user login.
+
+    If in RO state or in RW state with the help of tmpfs overlay,
+    syslog ERR messages are written, to help raise alerts.
+
+    Monit may be used to invoke it periodically, to help scan & fix and
+    report via syslog.
+
+"""
+
+import argparse
+import os
+import sys
+import syslog
+import subprocess
+
+TEST_LOG_FN = None
+UPPER_DIR = "/run/mount/upper"
+WORK_DIR = "/run/mount/work"
+MOUNTS_FILE = "/proc/mounts"
+
+def log_err(m):
+    print("Err: {}".format(m))
+    syslog.syslog(syslog.LOG_ERR, m)
+    if TEST_LOG_FN:
+        TEST_LOG_FN(syslog.LOG_ERR, m)
+
+
+def log_info(m):
+    print("Info: {}".format(m))
+    syslog.syslog(syslog.LOG_INFO, m)
+
+
+def log_debug(m):
+    print("debug: {}".format(m))
+    syslog.syslog(syslog.LOG_DEBUG, m)
+
+
+def test_rw(dirs): 
+    for d in dirs:
+        rw = os.access(d, os.W_OK)
+        if not rw:
+            log_err("{} dir is not RW".format(d))
+            return False
+        else:
+            log_debug("{} dir is RW".format(d))
+    return True
+
+
+def run_cmd(cmd):
+    proc = subprocess.run(cmd, shell=True, text=True, capture_output=True)
+    ret = proc.returncode
+    if ret:
+        log_err("failed: ret={} cmd={}".format(ret, cmd))
+    else:
+        log_info("ret={} cmd: {}".format(ret, cmd))
+
+    if proc.stdout:
+        log_info("stdout: {}".format(str(proc.stdout)))
+    if proc.stderr:
+        log_info("stderr: {}".format(str(proc.stderr)))
+    return ret
+
+
+def get_dname(path_name):
```

Comment:
>get_dname [](http://example.com/codeflow?start=4&length=9)

Could you use `os.path.basename(path_name)` ?

Reply:
os.path.basename(os.path.normpath(path_name))